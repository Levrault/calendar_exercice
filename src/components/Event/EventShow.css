.container {
  background-color: $palette-text-or-icon;
  border-radius: $space-s;
  bottom: resolve($space-l + $space-xs);
  box-shadow: 0 10px 20px rgba(0,0,0,.19), 0 6px 6px rgba(0,0,0,.23);
  left: -75px;
  min-width: 150px;
  padding: $space-m;
  position: absolute;
  z-index: $foreground;
}

.title {
  font-size: 1.1rem;
  text-align: center;
}

.buttons {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

