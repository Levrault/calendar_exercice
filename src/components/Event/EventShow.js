import React, { Component } from 'react';
import PropTypes from 'prop-types';
import EventButton from '../Button/EventButton';
import styles from './EventShow.css';

/**
 * Show/delete existing event
 * @class EventShow
 * @param {function} onDelete
 * @param {function} onCancel
 * @param {string} id
 * @param {string} color
 * @param {string} fontColor
 * @param {string} monthId
 * @param {string} name
 */
class EventShow extends Component {
  /**
   * Close form
   */
  handleCloseClick = () => {
    this.props.onCancel();
  }

  /**
   * Delete event and hide the form
   */
  handleDelete = () => {
    const { onDelete, id, monthId } = this.props;
    onDelete(id, monthId);
    this.handleCloseClick();
  }

  /**
  * Render
  * @returns {node}
  */
  render () {
    const { name, color, fontColor } = this.props;
    const fontStyle = {
      color: fontColor
    };

    return (
      <div className={styles.container} style={{ backgroundColor: color }}>
        <div className={styles.title} style={fontStyle}>
          {name}
        </div>

        <div className={styles.buttons}>
          <EventButton style={fontStyle} type="button" onClick={this.handleCloseClick}>
            <i className="material-icons">clear</i>
          </EventButton>

          <EventButton style={fontStyle} type="button" onClick={this.handleDelete}>
            <i className="material-icons">delete</i>
          </EventButton>
        </div>
      </div>
    );
  }
}

EventShow.propTypes = {
  onDelete: PropTypes.func.isRequired,
  onCancel: PropTypes.func.isRequired,
  id: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,
  fontColor: PropTypes.string.isRequired,
  monthId: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired
};

export default EventShow;
