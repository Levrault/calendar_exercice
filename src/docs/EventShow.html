<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: EventShow</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: EventShow</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>EventShow<span class="signature">(onDelete, onCancel, id, color, fontColor, monthId, name)</span><span class="type-signature"></span></h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    

    
    <h4 class="name" id="EventShow"><span class="type-signature"></span>new EventShow<span class="signature">(onDelete, onCancel, id, color, fontColor, monthId, name)</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    Show/delete existing event
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>onDelete</code></td>
            

            <td class="type">
            
                
<span class="param-type">function</span>


            
            </td>

            

            

            <td class="description last"></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>onCancel</code></td>
            

            <td class="type">
            
                
<span class="param-type">function</span>


            
            </td>

            

            

            <td class="description last"></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>id</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last"></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>color</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last"></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>fontColor</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last"></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>monthId</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last"></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>name</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last"></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Event_EventShow.js.html">components/Event/EventShow.js</a>, <a href="components_Event_EventShow.js.html#line6">line 6</a>
    </li></ul></dd>
    

    

    

    
</dl>


















    
    </div>

    

    

    

    

    

    

    
        <h3 class="subsection-title">Members</h3>

        
            
<h4 class="name" id="handleCloseClick"><span class="type-signature"></span>handleCloseClick<span class="type-signature"></span></h4>




<div class="description">
    Close form
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Event_EventShow.js.html">components/Event/EventShow.js</a>, <a href="components_Event_EventShow.js.html#line21">line 21</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="handleDelete"><span class="type-signature"></span>handleDelete<span class="type-signature"></span></h4>




<div class="description">
    Delete event and hide the form
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Event_EventShow.js.html">components/Event/EventShow.js</a>, <a href="components_Event_EventShow.js.html#line28">line 28</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    
    <h4 class="name" id="render"><span class="type-signature"></span>render<span class="signature">()</span><span class="type-signature"> &rarr; {node}</span></h4>
    

    



<div class="description">
    Render
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Event_EventShow.js.html">components/Event/EventShow.js</a>, <a href="components_Event_EventShow.js.html#line38">line 38</a>
    </li></ul></dd>
    

    

    

    
</dl>













<h5>Returns:</h5>

        


<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">node</span>


    </dd>
</dl>

    





        
    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="App.html">App</a></li><li><a href="Calendar.html">Calendar</a></li><li><a href="ColorPalette.html">ColorPalette</a></li><li><a href="Day.html">Day</a></li><li><a href="EventForm.html">EventForm</a></li><li><a href="EventShow.html">EventShow</a></li><li><a href="Month.html">Month</a></li><li><a href="YearForm.html">YearForm</a></li></ul><h3>Global</h3><ul><li><a href="global.html#About">About</a></li><li><a href="global.html#CalendarButton">CalendarButton</a></li><li><a href="global.html#checkValidServiceWorker">checkValidServiceWorker</a></li><li><a href="global.html#ColorButton">ColorButton</a></li><li><a href="global.html#colorChange">colorChange</a></li><li><a href="global.html#ColorInputField">ColorInputField</a></li><li><a href="global.html#dayReducer">dayReducer</a></li><li><a href="global.html#daySelector">daySelector</a></li><li><a href="global.html#deleteEvent">deleteEvent</a></li><li><a href="global.html#Error">Error</a></li><li><a href="global.html#EventButton">EventButton</a></li><li><a href="global.html#eventCreated">eventCreated</a></li><li><a href="global.html#eventDeleted">eventDeleted</a></li><li><a href="global.html#eventFormDeleteBegin">eventFormDeleteBegin</a></li><li><a href="global.html#eventFormDeleteError">eventFormDeleteError</a></li><li><a href="global.html#eventFormDeleteSuccess">eventFormDeleteSuccess</a></li><li><a href="global.html#eventFormReducer">eventFormReducer</a></li><li><a href="global.html#eventFormReset">eventFormReset</a></li><li><a href="global.html#eventFormSelector">eventFormSelector</a></li><li><a href="global.html#eventFormSubmitBegin">eventFormSubmitBegin</a></li><li><a href="global.html#eventFormSubmitError">eventFormSubmitError</a></li><li><a href="global.html#eventFormSubmitSuccess">eventFormSubmitSuccess</a></li><li><a href="global.html#eventsInit">eventsInit</a></li><li><a href="global.html#eventsReducer">eventsReducer</a></li><li><a href="global.html#eventsSelector">eventsSelector</a></li><li><a href="global.html#fetch">fetch</a></li><li><a href="global.html#fetchAll">fetchAll</a></li><li><a href="global.html#Link">Link</a></li><li><a href="global.html#Loader">Loader</a></li><li><a href="global.html#Main">Main</a></li><li><a href="global.html#mapStateToProps">mapStateToProps</a></li><li><a href="global.html#Message">Message</a></li><li><a href="global.html#Navbar">Navbar</a></li><li><a href="global.html#NavbarRouter">NavbarRouter</a></li><li><a href="global.html#onSelected">onSelected</a></li><li><a href="global.html#PageNotFound">PageNotFound</a></li><li><a href="global.html#post">post</a></li><li><a href="global.html#registerValidSW">registerValidSW</a></li><li><a href="global.html#render">render</a></li><li><a href="global.html#Title">Title</a></li><li><a href="global.html#unregister">unregister</a></li><li><a href="global.html#WeekHeader">WeekHeader</a></li><li><a href="global.html#yearFetchAllBegin">yearFetchAllBegin</a></li><li><a href="global.html#yearFetchAllError">yearFetchAllError</a></li><li><a href="global.html#yearFetchAllSuccess">yearFetchAllSuccess</a></li><li><a href="global.html#yearFetchBegin">yearFetchBegin</a></li><li><a href="global.html#yearFetchError">yearFetchError</a></li><li><a href="global.html#yearFetchSuccess">yearFetchSuccess</a></li><li><a href="global.html#yearFormReducer">yearFormReducer</a></li><li><a href="global.html#yearSelector">yearSelector</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.5.5</a> on Fri Sep 21 2018 15:33:44 GMT+0200 (CEST)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>