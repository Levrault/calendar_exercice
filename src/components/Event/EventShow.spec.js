import React from 'react';
import renderer from 'react-test-renderer';
import EventShow from './EventShow';
import { shallow } from '../../tools/test/enzyme';

const mock = {
  onDelete: jest.fn(),
  onCancel: jest.fn(),
  monthId: '0b54aaf1844478e4',
  name: 'hello',
  color: '#546E7A',
  fontColor: '#212121',
  day: 19,
  id: '8b058e6f439a8985'
};

describe('EventShow component', () => {
  it('should render without problem', () => {
    const component = renderer.create(
      <EventShow {...mock} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call onDelete when handleDelete is called', () => {
    const component = shallow(
      <EventShow {...mock} />,
    );

    component.instance().handleDelete();
    expect(component.instance().props.onDelete).toHaveBeenCalled();
    expect(component.instance().props.onCancel).toHaveBeenCalled();
  });

  it('should call onCancel when handleCloseClick is called', () => {
    const component = shallow(
      <EventShow {...mock} />,
    );

    component.instance().handleCloseClick();
    expect(component.instance().props.onCancel).toHaveBeenCalled();
  });
});
