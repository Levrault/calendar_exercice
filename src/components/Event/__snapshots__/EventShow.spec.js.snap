// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventShow component should render without problem 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#546E7A",
    }
  }
>
  <div
    style={
      Object {
        "color": "#212121",
      }
    }
  >
    hello
  </div>
  <div>
    <button
      className=""
      onClick={[Function]}
      style={
        Object {
          "color": "#212121",
        }
      }
      type="button"
    >
      <i
        className="material-icons"
      >
        clear
      </i>
    </button>
    <button
      className=""
      onClick={[Function]}
      style={
        Object {
          "color": "#212121",
        }
      }
      type="button"
    >
      <i
        className="material-icons"
      >
        delete
      </i>
    </button>
  </div>
</div>
`;
